src/odo/odoWrapper.ts

```diff
@@ -194,13 +194,12 @@ export class Odo {
         );
     }
 
-    public async analyze(currentFolderPath: string): Promise<AnalyzeResponse[]> {
+    public async alizerAnalyze(currentFolderPath: Uri): Promise<AlizerAnalyzeResponse[]> {
         const cliData: CliExitData = await this.execute(
-            new CommandText('odo', 'analyze -o json'),
-            currentFolderPath,
+            new CommandText('alizer', `devfile ${currentFolderPath.fsPath}`)
         );
-        const parse = JSON.parse(cliData.stdout) as AnalyzeResponse[];
-        return parse;
+        const parse = JSON.parse(cliData.stdout) as AlizerAnalyzeResponse[];
+        return [[...parse].shift()];
```

Comment:
Shift returns the first element. Why not just return the first element (or undefined if there is no element) and change the return type to `Promise<AlizerAnalyzeResponse>` ?